tests/api/utils/test_scheduler.py

```diff
@@ -610,6 +616,65 @@ async def test_update_schedule_failure_not_found(db: Session, scheduler: Schedul
     assert "Schedule not found" in str(excinfo.value)
 
 
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    # The function waits 2 seconds and the schedule runs every second for 4 seconds. So:
+    # For 1 concurrent job, the second and fourth jobs should be skipped resulting in 2 runs.
+    # For 2 concurrent jobs, the third job should be skipped resulting in 3 runs.
+    # For 3 concurrent jobs, no job should be skipped resulting in 4 runs.
+    "concurrency_limit,run_amount",
+    [(1, 2), (2, 3), (3, 4)],
+)
+@pytest.mark.parametrize(
+    "schedule_kind", [schemas.ScheduleKinds.job, schemas.ScheduleKinds.local_function]
+)
+async def test_schedule_job_concurrency_limit(
+    db: Session,
+    scheduler: Scheduler,
+    concurrency_limit: int,
+    run_amount: int,
+    schedule_kind: schemas.ScheduleKinds,
+):
+    global call_counter
+    call_counter = 0
+
+    now = datetime.now()
+    now_plus_4_seconds = now + timedelta(seconds=4)
+    cron_trigger = schemas.ScheduleCronTrigger(
+        second="*", start_date=now, end_date=now_plus_4_seconds
+    )
```

Comment:
Using now as start_date may create transient failure in CI when now is X.9xx and then the actual scheduling only happens when the time is already X+1, meaning you'll have the function scheduled 3 times instead of 4
we use to use now+1 second for start time, see `test_create_schedule`